internal/server/routes.go

```diff
@@ -140,41 +149,98 @@ func (s *Server) GenerateRoutes() Routes {
 type HandlerFunc[Req, Res any] func(c *gin.Context, req *Req) (Res, error)
 
 type route[Req, Res any] struct {
-	method              string
-	path                string
-	handler             HandlerFunc[Req, Res]
-	omitFromDocs        bool
-	omitFromTelemetry   bool
-	infraHeaderOptional bool
+	method                     string
+	path                       string
+	handler                    HandlerFunc[Req, Res]
+	omitFromDocs               bool
+	omitFromTelemetry          bool
+	infraVersionHeaderOptional bool
+	noAuthentication           bool
+	noOrgRequired              bool
+	requiredAuthorization      []string
```

Comment:
Not demonstrated in this PR, but to have routes set these values I think we'd declare these routes as static package-level variables. For example:

```
var createGrant = route{
    ...
    handler: createGrantHandler,
    requiredAuthorization: adminRole,
}

```

The method we call `API.CreateGrant` today would be renamed to `createGrantHandler`.

That way `GenerateRoutes` will continue to look almost identical to what it is today, but we can specify the authentication, authorization, and org requirement next to the handler.